### PR TITLE
Api v3 port

### PR DESCRIPTION
**Pull request by wojciechrauk**

### CODE REVIEW

File: apiaccess.py [hunk 1]
```diff
@@ -17,7 +17,30 @@ def _remote_headers(auth_string, case_id, language_model=None):
...
+    Args:
+        to_flatten (dict | None)
+
+    Returns:
+        dict | None: flattened dict, or None if given `to_flatten` is None
+    """
+    if to_flatten is None:
+        return None
+
+    items = []
```
Comment (Nummulit) [inline apiaccess.py]: I think we can do `items = {}` and then use `items.update(…)` and `items[new_key] = value` without using intermediate list :)

File: chat.py [hunk 2]
```diff
@@ -75,10 +73,6 @@ def parse_args():
     parser.add_argument("--model",
                         help="use non-standard Infermedica model/language, "
                              "e.g. infermedica-es")
-    # TODO: Check if `verbose` actually does anything.
```
Comment (Nummulit) [inline chat.py:78]: Removing a TODO is always good! 👍

File: conversation.py [hunk 3]
```diff
@@ -55,15 +55,17 @@ def read_age_sex():
...
     extract conditions found in text.
 
     Args:
+        age (dict): Patients age in {'value': int, 'unit': str} format.
+        sex (str): Patients sex.
         auth_string (str): Authentication string.
         case_id (str): Case ID.
         context (list): List previous complaints.
-        lanugage_model (str): Chosen language model.
+        language_model (str): Chosen language model.
```
Comment (Nummulit) [inline conversation.py:68]: Good catch, thanks 😅

**Review by Nummulit — CHANGES_REQUESTED** [review]

File: apiaccess.py [hunk 4]
```diff
@@ -37,10 +60,14 @@ def call_endpoint(endpoint, auth_string, request_spec, case_id,
     if request_spec:
         resp = requests.post(
             url,
+            params=_flatten_params_dict(params),
```
Comment (adam-ra) [inline apiaccess.py]: Why is this needed?

Comment (wojciechrauk) [reply apiaccess.py]: request.post or request.get set URL parameters from dictionary, so the correct dictionary here would be: `{"age.value": x, "age.unit": y}`, but in different endpoints this information about age is send via json (in post body), where structure is `{"age": {"value": x, "unit":y}}`. Alternatively we can omit this  `_fltten_params_dict` function and pass sth like ```call_endpoint(..., params={"age.value": age["value"], "age.unit": age["unit"]}``` What do you think?

File: apiaccess.py (hunk 1 above)
Comment (wojciechrauk) [reply apiaccess.py]: You were right :D I've changed it

**Review by Nummulit — APPROVED** [review]

File: apiaccess.py (hunk 4 above)
Comment (adam-ra) [reply apiaccess.py]: I think it would be better to do this directly using `age.unit` etc. This piece of code is intended to be a demo showing that our APIs are simple :sweat_smile: so I'd rather avoid suggesting potential users that you really need to implement a dict-flattening routine yourself to use the API.

Comment (wojciechrauk) [reply apiaccess.py]: okay, I've simplified it :)

Comment (wojciechrauk) [reply apiaccess.py]: Maybe squash my 4 commits into one?

**Review by adam-ra — APPROVED** [review]
